annotations/quadriga.py

```diff
@@ -275,3 +257,303 @@ def parse_response(raw_response):
+    
+    # Get the template part from the relationset's template
+    template_part = relationset.template.template_parts.first()
+    
+    # Get the subject node from the relation
+    subject_node = None
+    if template_part.source_node_type == 'CO':  # Specific concept
+        subject_node = top_relation.source_content_object
+    elif template_part.source_node_type == 'TP':  # Open concept
+        subject_node = top_relation.source_content_object
+    elif template_part.source_node_type == 'DT':  # Date
+        subject_node = top_relation.source_content_object
+    elif template_part.source_node_type == 'RE':  # Relation
+        subject_node = top_relation.source_content_object
+    
+    # Get the predicate node
+    predicate_node = top_relation.predicate
+    
+    # Get the object node from the relation
+    object_node = None
+    if template_part.object_node_type == 'CO':  # Specific concept
+        object_node = top_relation.object_content_object
+    elif template_part.object_node_type == 'TP':  # Open concept
+        object_node = top_relation.object_content_object
+    elif template_part.object_node_type == 'DT':  # Date
+        object_node = top_relation.object_content_object
+    elif template_part.object_node_type == 'RE':  # Relation
+        object_node = top_relation.object_content_object
+    
+    # Create keys for looking up node IDs
+    subj_key = None
+    obj_key = None
+    
+    if subject_node:
+        if hasattr(subject_node, 'source_content_type_id'):
+            subj_key = f"rel-{subject_node.id}-{subject_node.created.isoformat()}"
+        else:
+            subj_key = f"app-{subject_node.id}-{subject_node.created.isoformat()}"
+    
+    if object_node:
+        if hasattr(object_node, 'source_content_type_id'):
+            obj_key = f"rel-{object_node.id}-{object_node.created.isoformat()}"
+        else:
+            obj_key = f"app-{object_node.id}-{object_node.created.isoformat()}"
+    
+    # Check if using alternative expression format (node1 behavior:uri node2)
+    expression = relationset.template.expression
+    is_alternative_format = expression and ':' in expression and any(c.isalpha() for c in expression.split(':')[0])
+    
+    # Extract URI from alternative format if present
+    alt_format_uri = None
+    if is_alternative_format:
+        # Split on colon and get the URI part, strip any whitespace
+        alt_format_uri = expression.split(':')[1].strip()
+    
+    # Construct the default mapping based on template
+    default_mapping = {
+        "subject": {
+            "type": "REF",
+            "reference": node_mapping.get(subj_key, "0"),
+        },
+        "predicate": {
+            "type": "REF" if is_alternative_format else "URI",
+            "uri": alt_format_uri if is_alternative_format else predicate_node.interpretation.master.uri,
+        },
+        "object": {
+            "type": "REF",
+            "reference": node_mapping.get(obj_key, "0"),
+        }
```

Comment:
all three parts of the default mapping could be either a specific concept or a reference to a node in the graph.